domain/src/main/java/no/difi/meldingsutveksling/core/EDUCoreFactory.java

```diff
@@ -127,6 +144,127 @@ public EDUCore create(Message message, String senderOrgNr) {
+                avsmotType.setAmIhtype("0");
+            }
+            if ("mottaker".equals(k.getKorrespondanseparttype().value().toLowerCase())) {
+                avsmotType.setAmIhtype("1");
+            }
+            avsmotType.setAmAdmkort(k.getAdministrativEnhet());
+            avsmotType.setAmSbhinit(k.getSaksbehandler());
+            if (!jp.getAvskrivning().isEmpty()) {
+                Avskrivning avs = jp.getAvskrivning().get(0);
+                ofNullable(avs.getAvskrivningsmaate()).map(Avskrivningsmaate::value).ifPresent(avsmotType::setAmAvskm);
+                ofNullable(avs.getAvskrivningsdato()).map(XMLGregorianCalendar::toString).ifPresent(avsmotType::setAmAvskdato);
+                avsmotType.setAmAvsavdok(avs.getReferanseAvskrivesAvJournalpost());
+            }
+
+            journpostType.getAvsmot().add(avsmotType);
+        });
+
+        jp.getDokumentbeskrivelseAndDokumentobjekt().stream()
+                .filter(d -> d instanceof Dokumentbeskrivelse)
+                .forEach(b -> {
+                    Dokumentbeskrivelse db = (Dokumentbeskrivelse) b;
+                    db.getDokumentobjekt().forEach(dobj -> {
+                        DokumentType dokumentType = of.createDokumentType();
+                        String filename = dobj.getReferanseDokumentfil();
+                        dokumentType.setVeFilnavn(filename);
+                        dokumentType.setDbTittel(db.getTittel());
+                        dokumentType.setDlRnr(db.getDokumentnummer().toString());
+                        dokumentType.setDlType(db.getTilknyttetRegistreringSom().value());
+
+                        String[] split = dobj.getReferanseDokumentfil().split(".");
+                        String ext = Stream.of(split).reduce((p, e) -> e).orElse("pdf");
+                        dokumentType.setVeMimeType(getMimetype(ext));
+                        dokumentType.setVeVariant(dobj.getVariantformat().value());
+
+                        FilType filType = of.createFilType();
+                        try {
+                            filType.setBase64(getFileFromAsic(filename, asic));
+                        } catch (IOException e) {
+                            throw new MeldingsUtvekslingRuntimeException(String.format("Error getting file %s from ASiC", filename), e);
+                        }
+                        dokumentType.setFil(filType);
+
+                        journpostType.getDokument().add(dokumentType);
+                    });
+
+
+                });
+
+        MeldingType meldingType = of.createMeldingType();
+        meldingType.setJournpost(journpostType);
+        meldingType.setNoarksak(noarksakType);
+
+        // Further usage expects payload to be marshalled
+        EDUCoreConverter eduCoreConverter = new EDUCoreConverter();
+        eduCore.setPayload(eduCoreConverter.meldingTypeAsString(meldingType));
+
+        return eduCore;
+    }
+
+    private List<Dokumentobjekt> dokObjFromReg(Registrering r) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "dokObjFromReg" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AUnusedPrivateMethod)